app/api/routers/dex_market.py

```diff
@@ -166,144 +166,44 @@ def list_all_membership_order_book(
     is_buy = request_query.order_type == "buy"  # 相対注文が買い注文かどうか
     exchange_address = to_checksum_address(config.IBET_MEMBERSHIP_EXCHANGE_CONTRACT_ADDRESS)
 
+    query = (
+        session.query(
+            Order.order_id,
+            Order.amount,
+            Order.price,
+            Order.exchange_address,
+            Order.account_address,
+            func.sum(Agreement.amount)
+        ).
+        outerjoin(
+            Agreement,
+            and_(Order.unique_order_id == Agreement.unique_order_id,
+                 Agreement.status != AgreementStatus.CANCELED.value)  # 約定取消時に注文中状態に戻るため約定数量には取消分を含めない
+        ).
+        group_by(
+            Order.order_id,
+            Order.amount,
+            Order.price,
+            Order.exchange_address,
+            Order.account_address
+        ).
+        filter(Order.exchange_address == exchange_address).
+        filter(Order.token_address == token_address).
+        filter(Order.is_cancelled == False)  # 未キャンセル
+    )
+
+    if is_buy:  # 買注文
+        query = query.filter(Order.is_buy == False)
+    else:  # 売注文
+        query = query.filter(Order.is_buy == True)
+
```

Comment:
Refactored duplicated business logic.

Reply:
I think `exchange_agent_address` (newly added) in request_query should be used to filter `Order` data.

Reply:
Sorry. Thanks for pointing that out.
I have fixed the query and added a test case.